mac_linux_users/recommendation.py

```diff
@@ -0,0 +1,138 @@
+########################
+### RECOMMENDATIONS ####
+########################
+
+import argparse as arg
+import os
+import time
+import json
+from termcolor import *
+import colorama
+
+colorama.init()
+
+# Procedure for generating a match score for two images
+def matchScore(first, second, secondPath, json_fail, descExist):
+    '''
+    # Takes in certain arguments
+        1- First --> First meme's path/description
+        2- Second --> Second meme's path/description
+        3- secondPath --> Path of second image in case description does not exist
+        4- json_fail --> If first and second both are paths, this is set to True
+        5- descExist --> Flag showing that description for meme exist or not
+    '''
+    score = 0
+    #creating a match score based on common descriptions
+    if (json_fail): # Both first and second are paths
+        first = first.split('/')
```

Comment:
Utkarsh, I would like to suggest the use of a variable to set the delimiter and using that variable to split paths and other file names. This will eliminate the need for an extra file and effort of deleting and copying files.

- Platform detection can be done using a simple import : from sys import platform
now we could just add a one line check like
if platform == 'linux':
    sep = '/'
elif platform == 'win32':
    sep='\\'
and likewise.

The fileName function can then be tweaked to the previous version.
It would be great if you could look into this.

Reply:
Thanks Vikrant for suggesting the way to deal with this type of situations. I will review and make changes again.